shaping/input.go

```diff
@@ -33,3 +35,85 @@ type Input struct {
 	// Language is an identifier for the language of the text.
 	Language language.Language
 }
+
+// SplitByFontGlyphs split the runes from 'input' to several items, sharing the same
+// characteristics as 'input', expected for the `Face` which is set to
+// the first font among 'availableFonts' providing support for all the runes
+// in the item.
+// Runes supported by no fonts are mapped to the first element of 'availableFonts', which
+// must not be empty.
+// The 'Face' field of 'input' is ignored: only 'availableFaces' are consulted.
+// Rune coverage is obtained by calling the NominalGlyph() method of each font.
+func SplitByFontGlyphs(input Input, availableFaces []font.Face) []Input {
```

Comment:
I could see an argument that we should create some kind of intermediate type like:

```go
type Run struct {
    Text []rune
    Start, End int
}
```

We could easily embed this inside of the `Input` type, but use it as the input to this function.

Then again, we probably want to ensure that the other fields of the returned `[]Input`s all match (size, script, language, direction), so it probably isn't a net win.

I don't really think we should change this, but I'll leave this comment in case it sparks thoughts for others.